Update runtime, grab latest git commit.

F-i-f opened the pull request:
Update runtime to 23.08.
Update to latest git commit.
Drop SDL2 patch upstreamed.
Add C++ compilation fix for 22.08/23.08 SDKs.

Mailaender commented on net.sourceforge.DuneLegacy.yaml:
```diff
@@ -24,13 +24,12 @@ modules:
         url: https://git.code.sf.net/p/dunelegacy/code
-        tag: v0.96.4
-        commit: dede956fbd6f6f6bbee98a059c7edb677e67fc6d
+        commit: 6ea9ac96854daa8c75ba429e78dc6716b147e106
```
We need a released version. Ask upstream for a non-Windows release. https://dunelegacy.sourceforge.net/website/news.html#2022-12-27-00-00-00

F-i-f commented:
The project seems more or less abandoned, so I'm not sure I'll be able to get a release.
Would you accept a PR that:
- Fixes the linter errors
- Updates the runtime to 23.08
- But keeps the release at 0.96.4?

F-i-f committed:
Update runtime to 23.08.

Add a C++ patch to fix compilation with g++ from 23.08.

F-i-f committed:
Update appstream to latest standard.

Convert all links to https.
Add launchable and developer tags.
Add screenshot captions.

Mailaender commented:
Yes, that would be very much appreciated.

Mailaender commented:
bot, build

F-i-f commented:
Thank you for taking care of restarting the (transiently broken) build and merging.
